packages/smooth_app/lib/cards/category_cards/svg_safe_network.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:http/http.dart' as http;
+import 'package:smooth_app/cards/category_cards/asset_cache_helper.dart';
+import 'package:smooth_app/cards/category_cards/svg_async_asset.dart';
+import 'package:smooth_app/cards/category_cards/svg_cache.dart';
+import 'package:smooth_app/services/smooth_services.dart';
+
+/// Widget with async load of SVG network file.
+///
+/// We could use SvgPicture.network, but it sends tons of errors if there in no
+/// internet connection. That's why we download the data ourselves.
+class SvgSafeNetwork extends StatefulWidget {
+  const SvgSafeNetwork(this.helper);
+
+  final AssetCacheHelper helper;
+
+  @override
+  State<SvgSafeNetwork> createState() => _SvgSafeNetworkState();
+}
+
+class _SvgSafeNetworkState extends State<SvgSafeNetwork> {
+  late final Future<String> _loading = _load();
+
+  String get _url => widget.helper.url;
+
+  Future<String> _load() async {
+    String? cached = _networkCache[_url];
+    if (cached != null) {
+      return cached;
+    }
+    final http.Response response = await http.get(Uri.parse(_url));
+    _networkCache[_url] = cached = response.body;
+    return cached;
+  }
+
+  @override
+  Widget build(BuildContext context) => FutureBuilder<String>(
+        future: _loading,
+        builder: (BuildContext context, AsyncSnapshot<String> snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            if (snapshot.data != null) {
+              return SvgPicture.string(
+                snapshot.data!,
+                width: widget.helper.width,
+                height: widget.helper.height,
+                colorFilter: widget.helper.color == null
+                    ? null
+                    : ui.ColorFilter.mode(
+                        widget.helper.color!,
+                        ui.BlendMode.srcIn,
+                      ),
+                fit: BoxFit.contain,
+                semanticsLabel: SvgCache.getSemanticsLabel(context, _url),
+                placeholderBuilder: (BuildContext context) =>
+                    SvgAsyncAsset(widget.helper),
+              );
+            }
+          }
+          if (snapshot.error != null) {
+            final bool serverOrConnectionIssue = snapshot.error.toString() ==
+                "Failed host lookup: 'static.openfoodfacts.org'";
+            if (!serverOrConnectionIssue) {
+              Logs.e(
+                'Could not download "$_url"',
+                ex: snapshot.error,
+              );
+            }
+          }
+          return SvgAsyncAsset(widget.helper);
+        },
+      );
+}
+
+/// Network cache, with url as key and SVG data as value.
+final Map<String, String> _networkCache = <String, String>{};
```

Comment:
The drawback with this implementation of the cache is that we will use a lot of memory for this.
Could you instead use a `WeakReference`?

Reply:
> The drawback with this implementation of the cache is that we will use a lot of memory for this.

I wouldn't say "a lot", because those are only SVG files. Many of them, though.
Besides, I'm only replicating a process from the previous standard implementation (`SvgPicture.network`), that cached the downloaded data (`All network images are cached regardless of HTTP headers.`)

> Could you instead use a `WeakReference`?

I'm not very familiar with `WeakReference`, but from what I've just read that wouldn't look appropriate.
It's not because currently in the app there's no "nutriscore E" icon displayed that I should remove it from memory and re-download it again when needed.

Regarding the total size of svg files, you may also complain about the similar asset files. And in that case, they _always_ use memory, even when not used.

This is what I suggest in order to minimize the memory impacts of this cache:
1. instead of caching the svg, cache a zipped version of the svg => less memory used
2. just simply ignore the URLs and never download if we already have a cached version in assets => would be safe depending how often the svg are actually refreshed on the server side
3. just store a `bool` stating if the downloaded svg file is identical to the corresponding asset file
zip

For solutions 2 and 3, that would mean having the best possible svg files in `assets`, which will also solve #4617.
But actually that would mean much more memory used if we don't optimize the asset files...

Reply:
@g123k Do you have a preference among suggestions 1, 2 and 3?
There's also the solution to just add a `TODO` and consider that the PR is already an improvement, sentry-wise.

Reply:
Actually, the idea of `WeakReference` is that it will just free the content of the Map if the app lacks memory.
That's why just changing the type of the variable seems at the same a quick fix and a gain in terms of perfs.

But otherwise, I think option 2 is better, as, honestly, assets don't change that frequently.

Reply:
Still not convinced by the `WeakReference`, as
* it will be held in a `Map` (as opposed to a single variable - how can you say it's not used if the `Map` is used?)
* they say _The [WeakReference](https://api.flutter.dev/flutter/dart-core/WeakReference-class.html) constructor will reject any object that is not supported as an [Expando](https://api.flutter.dev/flutter/dart-core/Expando-class.html) key. ... Does not work on numbers, strings, ..._

But perhaps you meant that the whole `Map` would be a single `WeakReference`?

Reply:
Yes, my point was that to wrap the `Map` into a `WeakReference`

Reply:
@g123k Ok then, just pushed a version with `WeakReference`.
Could not test it unfortunately, as with the new version of flutter came a mandatory new version of xcode that my mac does not support...

Reply:
Even with the Android emulator?

Reply:
@g123k The thing is that I'm working on other flutter projects that are much easier to work on with an emulator (the android emulator is much slower than the ios one, for some reason).
I cannot stand this apple rat race anymore. My next computer will be a PC. Meanwhile I need to sort things out in my folders for a smooth transition.
That means I'll be less active in Smoothie, code-wise.

Reply:
@g123k I've just tested on an android emulator (finally I've downgraded my macos down to Catalina, thank you apple).
It does work: data is stored and reused.
And it does not work: the garbage collecting is very aggressive, and my weak reference is often getting cleared.

The `WeakReference` solution doesn't seem adapted to downloaded data.
And again, the "normal" implementation of `SvgPicture.network` does keep the data and never clears it, in favor of slow connectivity contexts.